Tools/ardupilotwaf/toolchain.py

```diff
@@ -24,14 +24,70 @@ def configure(cfg):
     OBJCOPY='objcopy',
 )
 
+def find_realexec_path(cfg, filename, path_list=[]):
+    if not filename:
+        return ''
+
+    if not path_list:
+        path_list = cfg.environ.get('PATH','').split(os.pathsep)
+
+    for dir in path_list:
+        path = os.path.abspath(os.path.expanduser(os.path.join(dir, filename)))
+
+        if os.path.isfile(path):
+            if os.path.islink(path):
+                realpath = os.path.realpath(path)
+
+                if filename not in os.path.basename(realpath):
+                    continue
+                else:
+                    return os.path.dirname(realpath)
+
+            else:
+                return os.path.dirname(path)
+
+    cfg.fatal('Could not find real exec path to %s in path_list %s:' % (filename, path_list))
```

Comment:
I said it too in the previous PR. I don't like it at all, but I think the only other way of doing this is if the user tell the build system where the toolchain is - and that doesn't seem very good either.